packages/teams-js/src/private/externalAppCardActions.ts

```diff
@@ -0,0 +1,169 @@
+    id: string;
+    data: Record<string, unknown>;
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface ICardActionsConfig {
+    enableImback: boolean;
+    enableInvoke: boolean;
+    enableDialog: boolean;
+    enableStageView: boolean;
+    enableSignIn: boolean;
+    enableO365Submit: boolean;
+  }
+
+  /**
+   *
+   * @hidden
+   * Error that can be thrown from IExternalAppCardActionService.handleActionSubmit
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface ActionSubmitError {
+    errorCode: ActionSubmitErrorCode;
+    message?: string;
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum ActionSubmitErrorCode {
+    INTERNAL_ERROR = 'INTERNAL_ERROR', // Generic error
+  }
+  /**
+   * @beta
+   * @hidden
+   * Delegates an Adaptive Card Action.Submit request to the host for the application with the provided app ID
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId ID of the application the request is intended for
+   * @param actionSubmitPayload The Adaptive Card Action.Submit payload
+   * @param cardActionsConfig The card actions configuration. This indicates which subtypes should be handled by this API
+   * @returns Promise that resolves when the request is completed and rejects with ActionSubmitError if the request fails
+   */
+  export function processActionSubmit(
+    appId: string,
+    actionSubmitPayload: IAdaptiveCardActionSubmit,
+    cardActionsConfig?: ICardActionsConfig,
+  ): Promise<void> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    return sendMessageToParentAsync<[boolean, ActionSubmitError]>('externalAppCardActions.processActionSubmit', [
+      appId,
+      actionSubmitPayload,
+      cardActionsConfig,
+    ]).then(([wasSuccessful, error]: [boolean, ActionSubmitError]) => {
+      if (!wasSuccessful) {
+        throw error;
+      }
+    });
+  }
+
+  /**
+   * @beta
+   * @hidden
+   * Delegates an Adaptive Card Action.OpenUrl request to the host for the application with the provided app ID
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId ID of the application the request is intended for
+   * @param url The URL to open
+   * @returns Promise that resolves to ActionOpenUrlType indicating the type of URL that was opened on success and rejects with ActionOpenUrlError if the request fails
+   */
+  export function processActionOpenUrl(appId: string, url: string): Promise<ActionOpenUrlType> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    // TODO: confirm no issues with deserialization
+    return sendMessageToParentAsync<[ActionOpenUrlError, ActionOpenUrlType]>('externalAppCardActions.processOpenUrl', [
```

Comment:
suggest to replace func value to be sent to Web Hub SDK as `externalAppCardActions.processActionOpenUrl`
cc @lakhveerk